File: marmoraria-app/src/views/layout_base.py
import flet as ft
from src.views.components.sidebar import Sidebar
from src.config import (
    COLOR_PRIMARY,
    COLOR_SECONDARY,
    COLOR_BACKGROUND,
    COLOR_WHITE,
    COLOR_WARNING,
    COLOR_TEXT,
)
from src.services import firebase_service


def LayoutBase(
    page: ft.Page,
    conteudo_principal,
    titulo="Central Granitos",
    subtitulo=None,
):
    """
    Layout base unificado
    Totalmente compatível com Flet 0.23.2
    """

    # ------------------------------------------------------
    # ---------------- STATUS DE CONEXÃO -------------------
    # ------------------------------------------------------
    try:
        conectado = firebase_service.verificar_conexao()
    except Exception:
        conectado = False

    # ------------------------------------------------------
    # ---------------- DETECÇÃO DE MOBILE ------------------
    # ------------------------------------------------------
    largura = page.width or 1024  # fallback seguro
    eh_mobile = largura < 768

    # ------------------------------------------------------
    # ---------------- DRAWER (MOBILE) ---------------------
    # ------------------------------------------------------
    drawer_mobile = ft.Container(
        content=Sidebar(page, is_mobile=True),
        width=260,
        bgcolor=COLOR_WHITE,
        padding=0,
    )

    def abrir_menu(e):
        page.drawer = drawer_mobile
        page.drawer.open = True
        page.update()

    # ------------------------------------------------------
    # ---------------- CONTEÚDO PRINCIPAL ------------------
    # ------------------------------------------------------
    view_wrapper = ft.Container(
        content=conteudo_principal,
        padding=20 if eh_mobile else 30,
        expand=True,
    )

    # ======================================================
    # ======================== MOBILE ======================
    # ======================================================
    if eh_mobile:

        app_bar = ft.AppBar(
            leading=ft.IconButton(
                icon="menu",
                icon_color=COLOR_WHITE,
                on_click=abrir_menu,
            ),
            title=ft.Column(
                spacing=0,
                horizontal_alignment=ft.CrossAxisAlignment.CENTER,
                controls=[
                    ft.Text(
                        titulo,
                        size=16,
                        weight=ft.FontWeight.BOLD,
                        color=COLOR_WHITE,
                    ),
                    ft.Text(
                        subtitulo,
                        size=11,
                        color=COLOR_WHITE,
                    ) if subtitulo else ft.Container(),
                ],
            ),
            center_title=True,
            bgcolor=COLOR_PRIMARY,
            elevation=0,
        )

        barra_status = ft.Container()
        if not conectado:
            barra_status = ft.Container(
                width=float("inf"),
                bgcolor=COLOR_WARNING,
                padding=8,
                content=ft.Row(
                    alignment=ft.MainAxisAlignment.CENTER,
                    spacing=6,
                    controls=[
                        ft.Icon(
                            name="wifi_off",
                            size=14,
                            color="black87",
                        ),
                        ft.Text(
                            "OFFLINE",
                            size=11,
                            weight=ft.FontWeight.BOLD,
                            color="black87",
                        ),
                    ],
                ),
            )

        # page.appbar e page.drawer são aplicados no main.py
        return ft.Container(
            expand=True,
            bgcolor=COLOR_BACKGROUND,
            data={
                "appbar": app_bar,
                "drawer": drawer_mobile,
            },
            content=ft.Column(
                spacing=0,
                controls=[
                    barra_status,
                    view_wrapper,
                ],
            ),
        )

    # ======================================================
    # ======================= DESKTOP ======================
    # ======================================================
    return ft.Container(
        expand=True,
        bgcolor=COLOR_BACKGROUND,
        content=ft.Row(
            spacing=0,
            expand=True,
            controls=[
                Sidebar(page, is_mobile=False),
                ft.Column(
                    expand=True,
                    spacing=0,
                    controls=[
                        view_wrapper,
                    ],
                ),
            ],
        ),
    )

File: lib/widgets/layout_base.dart
import 'package:flutter/material.dart';

import '../core/utils/constants.dart';
import '../features/widgets/sidebar.dart';
import '../core/services/firebase_service.dart';

class LayoutBase extends StatelessWidget {
  final Widget child;
  final String titulo;
  final String? subtitulo;

  const LayoutBase({
    super.key,
    required this.child,
    this.titulo = 'Central Granitos',
    this.subtitulo,
  });

  @override
  Widget build(BuildContext context) {
    final bool ehMobile = MediaQuery.of(context).size.width < 768;
    final bool conectado = FirebaseService.verificarConexao();

    // 🔹 Wrapper com padding + animação
    final viewWrapper = AnimatedOpacity(
      opacity: 1,
      duration: const Duration(milliseconds: 300),
      child: Padding(
        padding: EdgeInsets.all(ehMobile ? 20 : 30),
        child: child,
      ),
    );

    // 📱 MOBILE
    if (ehMobile) {
      return Scaffold(
        backgroundColor: COLOR_BACKGROUND,

        // 🟦 AppBar
        appBar: AppBar(
          backgroundColor: COLOR_PRIMARY,
          elevation: 0,
          centerTitle: true,
          leading: Builder(
            builder: (context) => IconButton(
              icon: const Icon(Icons.menu_rounded, color: COLOR_WHITE),
              onPressed: () => Scaffold.of(context).openDrawer(),
            ),
          ),
          title: Column(
            mainAxisSize: MainAxisSize.min,
            children: [
              Text(
                titulo,
                style: const TextStyle(
                  fontSize: 16,
                  fontWeight: FontWeight.bold,
                  color: COLOR_WHITE,
                ),
              ),
              if (subtitulo != null)
                Text(
                  subtitulo!,
                  style: const TextStyle(
                    fontSize: 11,
                    color: COLOR_WHITE,
                  ),
                ),
            ],
          ),
        ),

        // 📂 Drawer
        drawer: Drawer(
          backgroundColor: COLOR_WHITE,
          child: Sidebar(isMobile: true),
        ),

        // 🧠 Conteúdo
        body: Column(
          children: [
            // 🔴 Barra OFFLINE
            if (!conectado)
              Container(
                width: double.infinity,
                padding: const EdgeInsets.all(8),
                color: COLOR_WARNING,
                child: Row(
                  mainAxisAlignment: MainAxisAlignment.center,
                  children: const [
                    Icon(Icons.wifi_off, size: 14, color: Colors.black87),
                    SizedBox(width: 5),
                    Text(
                      'TRABALHANDO OFFLINE',
                      style: TextStyle(
                        fontSize: 11,
                        fontWeight: FontWeight.bold,
                        color: Colors.black87,
                      ),
                    ),
                  ],
                ),
              ),

            Expanded(child: viewWrapper),
          ],
        ),
      );
    }

    // 🖥️ DESKTOP
    return Scaffold(
      backgroundColor: COLOR_BACKGROUND,
      body: Row(
        children: [
          // Sidebar fixa
          const Sidebar(),

          // Conteúdo
          Expanded(
            child: Column(
              children: [
                Expanded(child: viewWrapper),
              ],
            ),
          ),
        ],
      ),
    );
  }
}
